client/app/_services/testsruns.service.js

```diff
@@ -64,6 +77,32 @@
 
         }
 
+        function filterLaunchers() {
+            _launchers.filter((launcher) => {
+                return getTestRunIndexByCiRunId(launcher.ciRunId, _lastResult.results) === -1;
+            })
+            storeLaunchers();
+        }
+
+        function checkForDestroyingLaunchersByTime() {
+            let dateNow = new Date();
+
+            _launchers.filter((launcher) => {
+                return launcher.shouldBeDestroyedAt - dateNow.getTime() > 0;
+            })
+            storeLaunchers();
+        }
+
+        function getTestRunIndexByCiRunId(ciRunId, arr) {
```

Comment:
Please use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex instead of this custom function